docs/design/core/metrics/prototype/MetricPublisherConfiguration.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics.publisher;
+
+import java.util.Collections;
+import java.util.List;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Configure the options to publish the metrics.
+ * <p>
+ * By default, SDK creates and uses only CloudWatch publisher with default options (Default credential chain
+ * and region chain).
+ * To use CloudWatch publisher with custom options or any other publishers, create a
+ * #PublisherConfiguration object and set it in the ClientOverrideConfiguration on the client.
```

Comment:
If I have an application that uses various libraries using the SDK, then is there a way to configure the metrics consistently for all uses of the SDK without needing to have each library explicitly do the right thing or configure it? 

We currently do it using an ExecutionInterceptor which can be loaded automatically if it is on the classpath. This has been handy for getting consistent instrumentation for all uses in an app without needing to worry about how the client was setup in the various libraries that are pulled in.

Reply:
Yes, you can set environment variables or system properties that can globally enable metrics.